Waveform: add track end markers

ronso0 opened the pull request:
![image](https://user-images.githubusercontent.com/5934199/94379983-e2810c00-0133-11eb-8758-8730eea6ee74.png)

https://bugs.launchpad.net/mixxx/+bug/1799576
This should help understand that no loops can be created (or resized) that would cross the track end, as well as the 
the protective loop move limitation in #3117 .

We could also use the end-of-track warning color for the post-roll triangles?

daschuer reviewed (changes requested):
Good Idea, Thank you. 
Unfortunately it does not work for short tracks where the lead in is visible as well.

ronso0 commented:
Good point, didn't think of that.
I replaced `if(preRollVisible) {} else {}` with `if(preRollVisible) {} if(postRollVisible) {}` and squashed all changes.
It now can paint both pre and post roll:
![image](https://user-images.githubusercontent.com/5934199/94435952-1b54cb80-019c-11eb-86fc-8e1aced9b2ea.png)

ronso0 commented:
And now I think it's okay to paint both with `SignalColor` instead of potentially causing confusion with another color.

daschuer commented:
Unfortunatly the clang-format is complaining: 

```
clang-format.............................................................Failed
- hook id: clang-format
- files were modified by this hook
[INFO] First pass: Reformatting added/changed lines...
[INFO] Reformatting src/waveform/renderers/waveformrendererpreroll.cpp [35-54, 73-94, 97-114]...
[INFO] Reformatting src/waveform/renderers/waveformwidgetrenderer.cpp [38-38, 137-138]...
[INFO] Reformatting src/waveform/renderers/waveformwidgetrenderer.h [80-80, 144-144]...
[INFO] Second pass: Breaking long added/changed lines...
```

ronso0 committed:
waveform: add track end markers

daschuer commented:
I don't mind in this case, because the surrounding code is written like that. So let's merge it.

In general I very much in favor that's all functions should be formated the same with no exceptions.
This improves readability IMHO.

daschuer reviewed (approved):
Lgtm
